### PR TITLE
Renamed Mailing Component entity

### PR DESCRIPTION
**Pull request by pradpnayak**
Overview
----------------------------------------
Mailing component was using entity name as 'Component' which overrides CRM/Core/DAO/Component.php since both were using same entity name. This PR renames the Mailing Component entity from 'Component' to 'MailingComponent'

### CODE REVIEW

**Comment by mattwire** [comment]
This looks like it makes sense.  @pradpnayak Was this breaking something specific?  @totten This is probably something you may understand/have a view on?

**Comment by eileenmcnaughton** [comment]
@colemanw @mattwire @seamuslee001 what do you think. I feel a bit nervous about this just because of risks of unknown interactions but I feel  like my concerns would be mitigated by keeping a copy of CRM_Mailing_DAO_Component that is an empty class extending the new one (& at some future point it could go)

File: CRM/Mailing/BAO/Component.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function __construct() {
    * @return CRM_Core_BAO_LocationType.
    */
   public static function retrieve(&$params, &$defaults) {
-    $component = new CRM_Mailing_DAO_Component();
+    $component = new CRM_Mailing_BAO_Component();
```
Comment (eileenmcnaughton) [inline CRM/Mailing/BAO/Component.php]: shouldn't we put the new DAO here - given the BAO needs renaming too but perhaps in a later iteration?

File: CRM/Mailing/BAO/Component.php [hunk 2]
```diff
@@ -71,7 +71,7 @@ public static function retrieve(&$params, &$defaults) {
    *   true if we found and updated the object, else false
    */
   public static function setIsActive($id, $is_active) {
-    return CRM_Core_DAO::setFieldValue('CRM_Mailing_DAO_Component', $id, 'is_active', $is_active);
+    return CRM_Core_DAO::setFieldValue('CRM_Mailing_BAO_Component', $id, 'is_active', $is_active);
```
Comment (eileenmcnaughton) [inline CRM/Mailing/BAO/Component.php]: ditto

File: CRM/Mailing/BAO/Component.php [hunk 3]
```diff
@@ -86,7 +86,7 @@ public static function setIsActive($id, $is_active) {
    */
   public static function add(&$params, $ids = array()) {
     $id = CRM_Utils_Array::value('id', $params, CRM_Utils_Array::value('id', $ids));
-    $component = new CRM_Mailing_DAO_Component();
+    $component = new CRM_Mailing_BAO_Component();
```
Comment (eileenmcnaughton) [inline CRM/Mailing/BAO/Component.php]: ditto

File: CRM/Mailing/BAO/Component.php (hunk 1 above)
Comment (colemanw) [reply CRM/Mailing/BAO/Component.php]: I agree. @pradpnayak can you update?

File: CRM/Mailing/BAO/Component.php (hunk 2 above)
Comment (colemanw) [reply CRM/Mailing/BAO/Component.php]: I agree. @pradpnayak can you update?

**Comment by colemanw** [comment]
I have updated the PR to also rename the BAO and to leave a deprecated copy of it to cover any backward-compat issue in extensions (although extensions really should not call BAOs directly for this reason).

**Comment by eileenmcnaughton** [comment]
test fails unrelated

**Pull request merged**